CollAction/Controllers/ProjectsController.cs

```diff
@@ -66,14 +67,24 @@ public ViewResult StartInfo()
         }
 
         // GET: Projects/Details/5
-        public async Task<IActionResult> Details(int? id)
+        public async Task<IActionResult> DetailsById(int? id)
         {
-            if (id == null)
+            var project =  await _service.GetProjectById(id); 
+            if (project == null)
             {
                 return NotFound();
             }
-
-            List<DisplayProjectViewModel> items = await DisplayProjectViewModel.GetViewModelsWhere(_context, p => p.Id == id && p.Status != ProjectStatus.Hidden && p.Status != ProjectStatus.Deleted);
+            string validUrlForProjectName = WebUtility.UrlEncode(project.Name);
+            if(String.IsNullOrEmpty(validUrlForProjectName))
+            {
+                return NotFound();
+            }
+            return LocalRedirect("~/projects/"+validUrlForProjectName+"/details");
```

Comment:
Using formatting might be a little more idiomatic, as in, LocalRedirect($"/Projects/{validUrlForProjectName}/Details").